code/ui/blocks/src/components/ArgsTable/ArgControl.tsx

```diff
@@ -65,9 +65,21 @@ export const ArgControl: FC<ArgControlProps> = ({ row, arg, updateArgs }) => {
 
   if (!control || control.disable) return <NoControl />;
 
+  // if the value is undefined and there is a default value, use it
+  let value = boxedValue?.value;
+  const defaultValue = row.table?.defaultValue;
+  if (value === undefined && defaultValue?.summary !== undefined) {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-implied-eval
+      value = Function(`return ${defaultValue.summary}`)();
+    } catch {
+      value = defaultValue.summary;
+    }
+  }
+
```

Comment:
@JReinhold 
this can be easily rectified by calling `updateArgs` after fetching the `defaultValue`.

something like:

```suggestion
  useEffect(() => {
    setTimeout(() => {
      updateArgs({ [key]: value });
    }, 0);
  }, [updateArgs, key, value]);

```

Reply:
I _think_ that would do the opposite of what I would expect. It would set the props (from args) to be the same as the component's internal defaults, instead of leaving them untouched so the internal defaults were actually used.